src/functions/sendCandidateResults/framework/status-updater.ts

```diff
@@ -1,21 +1,28 @@
+import axios from 'axios';
+import { inject, injectable } from 'inversify';
+import { TYPES } from './di/types';
+import { IConfigAdapter } from './adapter/config/config-adapter.interface';
+import { SubmissionOutcome } from '../domain/submission-outcome.model';
 
 export interface IStatusUpdater {
-  updateToAcceptedStatus(applicationId: number): void;
-
-  updateToFailedStatus(applicationId: number): void;
+  updateStatus(submissionOutcome: SubmissionOutcome): void;
 }
 
+@injectable()
 export class StatusUpdater implements IStatusUpdater {
-  constructor() {
 
-  }
+  private resultsBaseApiUrl: string ;
 
-  updateToAcceptedStatus(applicationId: number): void {
-    console.log(`Update test [${applicationId}] status to accepted`);
+  constructor(
+    @inject(TYPES.IConfigAdapter) private configAdapter: IConfigAdapter,
+  ) {
+    this.resultsBaseApiUrl = this.configAdapter.resultsBaseApiUrl;
   }
 
-  updateToFailedStatus(applicationId: number): void {
-    console.log(`Update test [${applicationId}] status to failed`);
+  updateStatus(submissionOutcome: SubmissionOutcome): void {
```

Comment:
Liked having those 2 functions because they gave more clarity for the possible statuses from the outside.

Reply:
The RequestScheduler tests now don't make any sense for example.